### DISCUSSION TITLE
i'm confused by `motion_estimate_filter` configuration comments

### DISCUSSION THREAD

**Discussion by shnizzedy** [FCP-INDI/C-PAC#1425] [Resolved] [answered]
How can options be mutually exclusive with required options?

https://github.com/FCP-INDI/C-PAC/blob/23d86fa845eed321f4ad88f12d91fffa5a74d17c/dev/docker_data/default_pipeline.yml#L775-L799

**Comment by shnizzedy** [comment]
Like this?

https://github.com/FCP-INDI/C-PAC/blob/d56996361a696e051893a2813c7789fdb2219e20/CPAC/pipeline/schema.py#L334-L383

| run | filter_type | breathing_rate_min | breathing_rate_max | center_frequency | filter_bandwidth | lowpass_cutoff |
| --- | --- | --- | --- | --- | --- | --- | 
| off | doesn't matter | doesn't matter | doesn't matter | doesn't matter | doesn't matter | doesn't matter | 
| on | notch | required | required | None | None | doesn't matter |
| on | notch | None | None | required | required | doesn't matter |
| on | lowpass | required | doesn't matter | doesn't matter | doesn't matter | None |
| on | lowpass | None | doesn't matter | doesn't matter | doesn't matter | required |

**Reply by sgiavasis** [reply]
Okay yeah, "required" is bad wording that wasn't updated. It was originally required. Then the ability to manually design the filters came in.

In the code, if the breathing rates are provided, the filter design attributes (center frequency, bandwidth, cutoff) are automatically configured. But if you provide these directly, you don't need the breathing rates.

**Reply by shnizzedy** [reply]
What if you provide both?

**Reply by sgiavasis** [reply]
The way it works now - it will always defer to breathing rate input, but only if both max and min are provided:

https://github.com/FCP-INDI/C-PAC/blob/master/CPAC/func_preproc/utils.py#L135-L147
https://github.com/FCP-INDI/C-PAC/blob/master/CPAC/func_preproc/utils.py#L167-L175

I think we should throw a warning telling the user that's what's happening, when it happens.
I just created an issue for this: https://github.com/FCP-INDI/C-PAC/issues/1430

**Comment by shnizzedy** [comment, marked as answer]
#1432: 

> https://github.com/FCP-INDI/C-PAC/blob/e8e6eec9bc0d6d3ff73c5b9940308b876e1be6b3/dev/docker_data/default_pipeline.yml#L891-L933
> 
> | run | filter_type | breathing_rate_min | breathing_rate_max | center_frequency | filter_bandwidth | lowpass_cutoff |
> | --- | --- | --- | --- | --- | --- | --- | 
> | **off** | doesn't matter | doesn't matter | doesn't matter | doesn't matter | doesn't matter | doesn't matter | 
> | on | **notch** | **required** | **required** | doesn't matter | doesn't matter | doesn't matter |
> | on | **notch** | **None** | **None** | **required** | **required** | doesn't matter |
> | on | **lowpass** | **required** | doesn't matter | doesn't matter | doesn't matter | doesn't matter |
> | on | **lowpass** | **None** | doesn't matter | doesn't matter | doesn't matter | **required** |
